tools/clang/test/CodeGenHLSL/mesh-val/msOversizeOutput.hlsl

```diff
@@ -0,0 +1,63 @@
+// RUN: %dxc -E main -T ms_6_5 %s | FileCheck %s
+
+// CHECK: vertex plus primitive output size is greater than 32768
+
+#define MAX_VERT 32
+#define MAX_PRIM 128
+#define NUM_THREADS 32
+struct MeshPerVertex {
+    float4 position : SV_Position;
+    float color[4] : COLOR;
+};
+
+struct MeshPerPrimitive {
+    float normal : NORMAL;
+    float4 malnor[16] : MALNOR;
+    int layer[4] : LAYER;
+};
+
+struct MeshPayload {
+    float normal;
+    float malnor;
+    int layer[4];
+};
+
+[numthreads(NUM_THREADS, 1, 1)]
+[outputtopology("triangle")]
+void main(
+            out indices uint3 primIndices[MAX_PRIM],
+            out vertices MeshPerVertex verts[MAX_VERT],
+            out primitives MeshPerPrimitive prims[MAX_PRIM],
+            in payload MeshPayload mpl,
+            in uint tig : SV_GroupIndex,
+            in uint vid : SV_ViewID
+         )
+{
+    SetMeshOutputCounts(MAX_VERT, MAX_PRIM);
+    MeshPerVertex ov;
+    if (vid % 2) {
+        ov.position = float4(4.0,5.0,6.0,7.0);
+        ov.color[0] = 4.0;
+        ov.color[1] = 5.0;
+        ov.color[2] = 6.0;
+        ov.color[3] = 7.0;
+    } else {
+        ov.position = float4(14.0,15.0,16.0,17.0);
+        ov.color[0] = 14.0;
+        ov.color[1] = 15.0;
+        ov.color[2] = 16.0;
+        ov.color[3] = 17.0;
+    }
+    if (tig % 3) {
```

Comment:
Although this is a test for output size limit, this appears to be a pattern repeated in other tests (including basic valid test mesh.hlsl), where I think this code should read `if (0 == (tig % 3))`.  Otherwise you have a write collision with different values to the same output array element for primIndices[] and prims[].  Same comment applies to `msOversizePayloadOutput.hlsl` test added here as well.

Additional issues:
- not all primitive outputs are initialized, since `tig / 3` for max `tig` of 31 is 10.  In this case, we set num prims to 128 to test output size max, but this is even the case for the mesh.hlsl unit test, where `MAX_PRIM` is 16.
- vertex index `tig + 2` will be out of vertex array bounds for the last primitive here.

It's possible that more rigorous compiler checks or validation in the future could fail on these shaders for reasons other than what was being tested here.

It's probably better to set the output counts correctly, and even to use `if (tig < MAX_PRIMS)`, `uint3 indices = (tig * 3) + uint3(0, 1, 2);`, then index primitives with tig: `primIndices[tig] = indices;`.  I think this would be a better pattern to model for future tests.

People may look at unit tests to see how the feature is used, so it's best not to establish undesired patterns here.